### PR TITLE
[Truffle] Use private maven repo instead of $HOME/.m2

### PR DESCRIPTION
**Pull request by jtulach**
I need to use local repository when building JRuby, otherwise parallel builds accessing $HOME/.m2 influence each other and cause random failures.

This change depends on `mx` changeset https://github.com/jtulach/mx/commit/9944309377dd671b6de3e5b8d8b9a014b34ee35e named "Add --repo parameter to mx maven-install to control where to install the JARs to".

### CODE REVIEW

File: mx.jruby/mx_jruby.py [hunk 1]
```diff
@@ -81,22 +82,22 @@ def build(self):
...
+        mx.run_mx(['maven-install', '--repo', mavenDir], suite=truffle)
 
         open(os.path.join(rubyDir, 'VERSION'), 'w').write('graal-vm\n')
 
         # Build jruby-truffle
 
-        mx.run(['find', '.'], nonZeroIsFatal=False, cwd=rubyDir)
-        mx.run_maven(['--version'], nonZeroIsFatal=False, cwd=rubyDir)
+        #mx.run(['find', '.'], nonZeroIsFatal=False, cwd=rubyDir)
+        mx.run_maven(['--version', '-Dmaven.repo.local=' + mavenDir], nonZeroIsFatal=False, cwd=rubyDir)
```
Comment (eregon) [inline mx.jruby/mx_jruby.py]: This should be saved in a local variable, similar to the mx patch.

Comment (jtulach) [inline mx.jruby/mx_jruby.py]: I am not sure what you mean...

File: mx.jruby/mx_jruby.py [hunk 2]
```diff
@@ -81,22 +82,22 @@ def build(self):
...
-        mx.run_mx(['maven-install'], suite=truffle)
+        mx.run_mx(['maven-install', '--repo', mavenDir], suite=truffle)
 
         open(os.path.join(rubyDir, 'VERSION'), 'w').write('graal-vm\n')
 
         # Build jruby-truffle
 
-        mx.run(['find', '.'], nonZeroIsFatal=False, cwd=rubyDir)
-        mx.run_maven(['--version'], nonZeroIsFatal=False, cwd=rubyDir)
+        #mx.run(['find', '.'], nonZeroIsFatal=False, cwd=rubyDir)
```
Comment (eregon) [inline mx.jruby/mx_jruby.py]: Please remove code instead of commenting it, git can always find it back should we need it.

**Comment by jtulach** [comment]
The necessary [mx commit](https://github.com/graalvm/mx/commit/9944309377dd671b6de3e5b8d8b9a014b34ee35e) is in - you can merge once you feel the change is OK.

**Comment by eregon** [comment]
Looks great now, thanks!

**Pull request merged**